client/hotrod-client-new/src/main/java/org/infinispan/client/hotrod/impl/operations/AbstractKeyHolderOperation.java

```diff
@@ -0,0 +1,43 @@
+package org.infinispan.client.hotrod.impl.operations;
+
+import org.infinispan.api.common.CacheEntry;
+import org.infinispan.client.hotrod.impl.InternalRemoteCache;
+import org.infinispan.client.hotrod.impl.VersionedOperationResponse;
+import org.infinispan.client.hotrod.impl.protocol.Codec;
+import org.infinispan.client.hotrod.impl.protocol.HotRodConstants;
+
+import io.netty.buffer.ByteBuf;
+
+public abstract class AbstractKeyHolderOperation<K, V> extends AbstractKeyOperation<V> {
```

Comment:
I've added this class to the hierarchy to hold the key for the specific operations in the new API. I didn't want to change `AbstractKeyOperation` as that seems to affect many more classes.

Let me know if you can think of another approach for this.

Reply:
I don't think this is required at all. I believe I sent it to you via message, but I highly recommend just changing the affected commands to just return a `MetadataValue` just like `GetWithMetadataOperation` does today (ignoring the retry part). Then the RemoteCacheImpl can convert the `MetadataValue` to a `CacheEntry` in a `thenApply` as it already has the key from the request. Then we don't need to add this class.